rs/messaging/src/routing/stream_builder/tests.rs

```diff
@@ -760,6 +764,250 @@ fn build_streams_with_best_effort_messages() {
     }
 }
 
+/// Given a stream with some (potentially zero) initial refunds and canister
+/// messages, tests that `build_streams()` respects the various limits when
+/// routing additional refunds and canister messages:
+///
+///  * Maximum stream size (set to 10 in this test).
+///  * Maximum refunds per stream (half the maximum stream size).
+///  * For system subnets, at most `2 * system_subnet_stream_msg_limit` canister
+///    messages per stream.
+///  * None of the above limits apply to the loopback stream.
+///
+/// The negative lower bounds in the strategies (that are then clamped to zero)
+/// aim to make it less likely that we start off with an already full stream; or
+/// that the stream fills up before any canister messages can be routed.
+#[test_strategy::proptest]
+fn build_streams_with_refunds(
+    // Stream may have up to 5 initial refunds.
+    #[strategy(-5..5isize)] initial_refunds: isize,
+    // And up to 10 total initial messages.
+    #[strategy(-5..#initial_refunds.max(0))] initial_messages: isize,
```

Comment:
It took me a while to make the link between the `total` in the comment and how we end up with a maximum of 10 messages. Maybe extend the comment to say something like this?

```suggestion
    // And up to 10 total initial messages (including refunds from above).
    #[strategy(-5..#initial_refunds.max(0))] initial_messages: isize,
```